File: src/codes/Users_Manual/controller_2.rs
# use autd3::prelude::*;
# use autd3::gain::IntoBoxedGain;
# use std::collections::HashMap;
# fn main() -> Result<(), Box<dyn std::error::Error>> {
#     let mut autd = Controller::open(
#         [AUTD3::default(), AUTD3::default()],
#         autd3::link::Nop::new(),
#     )?;
#     let x = 0.;
#     let y = 0.;
#     let z = 0.;
autd.group_send(
    |dev| match dev.idx() {
        0 => Some("focus"),
        1 => Some("null"),
        _ => None,
    },
    HashMap::from([
        (
            "focus",
            Focus {
                pos: Point3::new(x, y, z),
                option: Default::default(),
            }
            .into_boxed(),
        ),
        ("null", Null {}.into_boxed()),
    ]),
)?;
#     Ok(())
# }

File: src/codes/Users_Manual/controller_2.cpp
//~#include<chrono>
//~#include<autd3.hpp>
//~#include<autd3/link/nop.hpp>
//~int main() {
//~using namespace autd3;
//~auto autd =
//~Controller::open({AUTD3{}}, link::Nop{});
//~const auto x = 0.0;
//~const auto y = 0.0;
//~const auto z = 0.0;
autd.group_send(
    [](const Device& dev) -> std::optional<const char*> {
      if (dev.idx() == 0) {
        return "null";
      } else if (dev.idx() == 1) {
        return "focus";
      } else {
        return std::nullopt;
      }
    },
    std::unordered_map<const char*, std::shared_ptr<driver::Datagram>>{
        {"focus",
         std::make_shared<Focus>(Focus(Point3(x, y, z), FocusOption{}))},
        {"null", std::make_shared<Null>()}});
//~return 0; }